src/main/java/io/github/nucleuspowered/nucleus/modules/playerinfo/commands/NearCommand.java

```diff
@@ -0,0 +1,125 @@
+import io.github.nucleuspowered.nucleus.modules.core.datamodules.CoreUserDataModule;
+import io.github.nucleuspowered.nucleus.modules.playerinfo.config.PlayerInfoConfigAdapter;
+import org.spongepowered.api.command.CommandResult;
+import org.spongepowered.api.command.CommandSource;
+import org.spongepowered.api.command.args.CommandContext;
+import org.spongepowered.api.command.args.CommandElement;
+import org.spongepowered.api.command.args.GenericArguments;
+import org.spongepowered.api.entity.living.player.Player;
+import org.spongepowered.api.entity.living.player.User;
+import org.spongepowered.api.event.cause.Cause;
+import org.spongepowered.api.text.Text;
+import org.spongepowered.api.text.action.HoverAction;
+import org.spongepowered.api.text.action.TextActions;
+import org.spongepowered.api.util.annotation.NonnullByDefault;
+import org.spongepowered.api.world.Location;
+import org.spongepowered.api.world.World;
+
+import java.util.*;
+import java.util.function.Supplier;
+import java.util.stream.Collectors;
+
+@Permissions(suggestedLevel = SuggestedLevel.USER, supportsOthers = true)
+@RegisterCommand("near")
+@EssentialsEquivalent("near")
+@NonnullByDefault
+public class NearCommand extends AbstractCommand<CommandSource> implements Reloadable {
+
+    private final String playerKey = "subject";
+    private final String radiusKey = "radius";
+    private int maxRadius;
+
+    @Override
+    protected Map<String, PermissionInformation> permissionSuffixesToRegister() {
+        Map<String, PermissionInformation> pi = super.permissionSuffixesToRegister();
+        pi.put("others", PermissionInformation.getWithTranslation("permission.near.others", SuggestedLevel.MOD));
+        pi.put("maxexempt", PermissionInformation.getWithTranslation("permission.near.maxexempt", SuggestedLevel.MOD));
+        return pi;
+    }
+
+    @Override
+    public CommandElement[] getArguments() {
+        return new CommandElement[] {
+                GenericArguments.optionalWeak(requirePermissionArg(GenericArguments.user(Text.of(this.playerKey)), this.permissions.getOthers())),
+                GenericArguments.optionalWeak(GenericArguments.integer(Text.of(this.radiusKey)))
+        };
+    }
+
+    //near radius
+    //near player radius
+    @Override
+    protected CommandResult executeCommand(CommandSource src, CommandContext args, Cause cause) throws Exception {
+        final User user = this.getUserFromArgs(User.class, src, this.playerKey, args);
+        final Location<World> location;
+        if (user.isOnline()) {
+            location = user.getPlayer().get().getLocation();
+        } else {
+            Supplier<ReturnMessageException> returnMessageExceptionSupplier = () -> ReturnMessageException.fromKey("command.near.location.nolocation", user.getName());
+            location = Nucleus.getNucleus().getUserDataManager().get(user)
+                    .orElseThrow(returnMessageExceptionSupplier)
```

Comment:
I realise that you probably copied this from some legacy code I have, but since then, I realised the wonders of flatMap - will that work here instead of two orElseThrow methods?

Reply:
I'm not sure on the usage of flatMap

Reply:
`Optional#flatMap` operates on the contents of the optional if it exists, and expects an `Optional` to be returned - but then smartly unwraps the `Optional` so that you don't have `Optional<Optional<blah>>` returned. It is effectively saying the same as (where func(x) transforms x into an `Optional<Y>`):

```java
Optional<X> x = ...;
Optional<Y> y = optionalX.map(x -> func(x)) // currently Optional<Optional<Y>>
    .orElse(Optional::empty); // makes it Optional<Y>

// better to do
Optional<Y> y = optionalX.flatMap(x -> func(x)) // Optional<Y>
```

So, in this case,

```java
Nucleus.getNucleus().getUserDataManager().get(user)
     .flatMap(x -> x.get(CoreUserDataModule.class).getLogoutLocation())
     .orElseThrow(...);
```